### PR TITLE
feature/PI/raspbian-dockerfile-taskrunner

### PR DESCRIPTION
**Pull request by colton-smith**
This feature branch reuses a Dockerfile and some parts of the readme created by Mark Duffet for CompV at: https://github.com/ParadigmHyperloop/comp5-software/pull/7

```
Pidaq/docker/README.md - lower level instructions including docker commands
Pidaq/README.md - instructions for pi development and using taskrunner
```

Features:
* Dockerfile for raspbian with clang configuration
* Updated the readmes and renamed some semantic items to fit testing-software
* Vscode task runner + "Tasks" extension by actboy168 configured for one-click compile and run for raspberry pi architecture
* echoTest task to test the functionality of the task runner
* BuildPiImage task to build the PI container image, 15 minutes the first time and then ~3 seconds when the large items are cached 
* compileRunPiHelloWorld task to compile and run a "Hello World" c++ program in the raspbian image

### CODE REVIEW

File: Pidaq/README.md [hunk 1]
```diff
@@ -1,8 +1,20 @@
...
 This directory contains all of the code that will run on the raspberry pi. The pi will receive and parse CAN messages containing raw telemetry, apply calculations,  and create a packet(s) containing either a JSON or protobuf message to send to the control laptop via ethernet (UDP)
-*** 
-## Examples
+ 
+# Development
+We will be using docker to cross compile c++ for the raspberry pi
 
-Includes relevant examples 
-***
+Ensure docker is installed and running on your machine. Windows 10 Home does not have proper support for docker, however MUN offers a free upgrade to windows 10 education which is fully featured. The update is easy, unintrusive, and tasks less than 1 hour.
```
Comment (Burke-Daniel) [inline Pidaq/README.md]: Lil typo here, "tasks" should be "takes"

Comment (colton-smith) [reply Pidaq/README.md]: Fixed

File: Pidaq/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+!important vscode/tasks.json
```
Comment (Burke-Daniel) [inline Pidaq/.gitignore]: Should this be .vscode/tasks.json? I think that directory is still getting gitignored, just noticed this when following the instructions on the README

Comment (colton-smith) [reply Pidaq/.gitignore]: You're correct, fixed

**Review by Burke-Daniel — APPROVED** [review]
Works perfect on my machine now. Good to merge.

**Pull request merged**